### DISCUSSION TITLE
Cant get my Halmet to link to Pi5 Hotspot

### DISCUSSION THREAD

**Discussion by ESchlotfeldt** [hatlabs/discussions#122] [Q&A] [answered]
Ive tried everything re the above, without success:
- Hotspot: set up at 2.4GHz & connect automatically with priority is enabled
- SignalK: set up is as required (SSL not enabled, mDNS is enabled)
- I get the following messages in Platformio Terminal:
<img width="582" alt="image" src="https://github.com/hatlabs/discussions/assets/106944628/85130ff1-27e5-49c0-862e-bd7a1a0b2ee2">

**Comment by mairas** [comment, marked as answer]
For some inexplicable reason, the Raspberry Pi folks have on their latest OS version made the default hotspot settings to use WEP encryption which has been broken already for two decades. The Arduino ESP32 core, sensibly, requires WPA encryption at the very minimum, preventing connecting to the Pi hotspot.

I investigated the issue a while back and couldn't find an easy way to switch to WPA on the default Bookworm network stack. I gave up and downgraded the encryption requirement on SensESP v3 branch.

I quickly adapted the HALMET example firmware to SensESP v3. Try this: 

https://github.com/hatlabs/HALMET-example-firmware/tree/dev-3

**Reply by ESchlotfeldt** [reply]
Thank you - I've done this and it now seems to be able log onto the hotspot, so that's great. 

However, one problem seems to lead to another and now it doesn't seem to generate the requisite device access request in SignalK. I get the following error terminal messages:
<img width="485" alt="image" src="https://github.com/hatlabs/discussions/assets/106944628/f9720c32-1c97-4671-83a9-823e1b5305b0">

Any thoughts?

**Reply by ESchlotfeldt** [reply]
OK, it now generates the device access request, but when I type "never" and click approve, nothing happens and it keeps waiting for approval.

<img width="510" alt="image" src="https://github.com/hatlabs/discussions/assets/106944628/bf169ba8-8aa9-4143-afe3-81903a2a203d">

**Reply by ESchlotfeldt** [reply]
OK - i didn't use caps for NEVER. Seems to work now.

**Reply by mairas** [reply]
Reported as a signalk-server issue: https://github.com/SignalK/signalk-server/issues/1739

**Comment by ESchlotfeldt** [comment]
Great thanks! I’ll try that.On 17 May 2024, at 23:41, Matti Airas ***@***.***> wrote:﻿
For some inexplicable reason, the Raspberry Pi folks have on their latest OS version made the default hotspot settings to use WEP encryption which has been broken already for two decades. The Arduino ESP32 core, sensibly, requires WPA encryption at the very minimum, preventing connecting to the Pi hotspot.
I investigated the issue a while back and couldn't find an easy way to switch to WPA on the default Bookworm network stack. I gave up and downgraded the encryption requirement on SensESP v3 branch.
I quickly adapted the HALMET example firmware to SensESP v3. Try this:
https://github.com/hatlabs/HALMET-example-firmware/tree/dev-3
